media-tv/mythtv: Bug fix #665928 #693648 #674216 #693648 #695688 #695684

thebitpit opened the pull request:
Bump revision to `mythtv-29.1-r2` because bug #693648 relaxed restrictions to build with current version of `media-libs/libvpx`.
`

juippis commented:
Yes yes, it's on the queue. There are many other PRs waiting as well...
can't test this yet, but eye-sighting through I'll leave some notes:

juippis commented on media-tv/mythtv/Manifest:
```diff
@@ -1,2 +1,2 @@
-DIST mythtv-29.1-r1.tar.gz 105657217 BLAKE2B 9d42464b0c75c2d3a359b528e3f7917cfb281f48451167afc6517caa9702da44fce2d48566fd902ec08af35117232113b4d499e2c9a3cf2ad2634d91cf532ca7 SHA512 0eadc37661d9fa93a63c0f425cd43d43533889b0afcea1478b2645e083ba6c96eebf2eec088db1ed6f0a0727c03b7c90c0c18b83c194d630dda3bac45bca5d87
+DIST mythtv-29.1-r2.tar.gz 105657217 BLAKE2B 9d42464b0c75c2d3a359b528e3f7917cfb281f48451167afc6517caa9702da44fce2d48566fd902ec08af35117232113b4d499e2c9a3cf2ad2634d91cf532ca7 SHA512 0eadc37661d9fa93a63c0f425cd43d43533889b0afcea1478b2645e083ba6c96eebf2eec088db1ed6f0a0727c03b7c90c0c18b83c194d630dda3bac45bca5d87
```
This should be `${P}`, as you can see, the manifest hash is the same, so users are downloading the same massive file again. Also it stresses the mirrors. ALSO dont use `git mv` but you need to separate commits for addition and removal. But please, for now, make all 29.1 ebuilds use the -r1.tar.gz so it doesnt need to be downloaded and re-hosted again.

thebitpit replied:
make all 29.1 ebuilds use the -r1.tar.gz

thebitpit replied:
Restored mythtv-29.1-r1.ebuild in case someone needs it.

juippis commented on media-tv/mythtv/metadata.xml:
```diff
@@ -16,7 +16,7 @@
 		<flag name="cec">Allows control of CEC enabled TVs via HDMI</flag>
 		<flag name="ceton">Ceton InfiniTV 4 a CableCARD-enabled tuner support</flag>
-		<flag name="crystalhd">Broadcom CrystalHD hardware support</flag>
+		<flag name="crystalhd">Broadcom CrystalHD hardware support needs libcrystalhd to build</flag>
```
This **needs** to be removed. Its not functional. We dont serve dependencies that can be got from an overlay, or "if you happened to install this in 2010".

thebitpit replied:
Removed from both versions

juippis commented on media-tv/mythtv/mythtv-29.1-r2.ebuild:
```diff
@@ -63,17 +62,14 @@ COMMON="
 	fftw? ( sci-libs/fftw:3.0= )
 	hls? (
-		<media-libs/libvpx-1.7.0:=
+		media-libs/libvpx:=
```
I hope you are sure that it works, didnt I ask you to check this during the initial review? :)

thebitpit replied:
mythtv-29.1-r1 (original) is not compatible with the stable tree. Change to <media-libs/libvpx-1.8.0:= and it builds. Put same restriction in mythtv-30.0_p20190808.

juippis commented on media-tv/mythtv/mythtv-29.1-r2.ebuild:
```diff
@@ -62,9 +62,6 @@ COMMON="
 		sys-fs/udisks:2
 	)
 	cec? ( dev-libs/libcec )
-	dvb? (
```
Shouldn't this be removed from the IUSE list as well?

I think these commits look good, just the commit summaries need some renaming and they are a bit all over the place. Anyway I can edit those, just need an answer what would happen if dvb was removed from IUSE.

thebitpit replied:
Do not remove dvb use flag. The dvb use flag is passed to mythtv. The dependency is no longer needed as headers are in kernel now.

juippis commented:
I cant get this to apply cleanly in ::gentoo. Also I noticed you used `git mv` on your first commit, you should make a new copy instead and remove previous ebuild in its own commit.

I think you should start clean, and make 2-3 commits instead of 8, where you fix all the bugs in a single revision bump with new -r1 file. Make separate commits for 29.1 and 30.0. Then a new commit to clean the old files. Thats it.

Sorry this became so messy :\

thebitpit commented:
> I think you should start clean, and make 2-3 commits instead of 8, where you fix all the bugs in a single revision bump with new -r1 file. Make separate commits for 29.1 and 30.0. Then a new commit to clean the old files. Thats it.
> 
This is my interpretation of what is needed.
* commit of changes to 29.1 with one fix to make it compile in  ::gentoo tree. (includes changes to metadata.xml)
* commit of new 29.1-r1 with all fixes in PR for 29.1 (includes changes to Manifest)
* commit of changes to 30.0_p20190808 with all fixes in PR for 30.0_p20190808 
-- or commit for new 30.0-r1_p20190808 and changes to Manifest?)
-- Which is correct?

juippis commented:
Thanks, a lot cleaner now and it applies to ::gentoo git tree. I'll do few test runs and merge if its good, or report here if it breaks for me.

juippis commented:
Btw, to replace `egethome` and drop `user.eclass`, you can just check how the function is contructed from https://gitweb.gentoo.org/repo/gentoo.git/tree/eclass/user.eclass

But change it on next version bump. We'd like everyone to migrate to acct- method anyway. 

I still need to file bugs about the find -exec things, but need to test them first ;)

juippis commented:
```
* ERROR: media-tv/mythtv-30.0_p20190808-r1::testworld failed (prepare phase):
 *   The source directory '/tmp/portage/media-tv/mythtv-30.0_p20190808-r1/work/mythtv-30.0_p20190808-r1/mythtv' doesn't exist
 *
 * Call stack:
 *            ebuild.sh, line  773:  Called __ebuild_main 'prepare'
 *   phase-functions.sh, line 1072:  Called __dyn_prepare
 *   phase-functions.sh, line  384:  Called die
 * The specific snippet of code:
 *              die "The source directory '${S}' doesn't exist"
 *
 * If you need support, post the output of `emerge --info '=media-tv/mythtv-30.0_p20190808-r1::testworld'`,
 * the complete build log and the output of `emerge -pqv '=media-tv/mythtv-30.0_p20190808-r1::testworld'`.
 * The complete build log is located at '/tmp/portage/media-tv/mythtv-30.0_p20190808-r1/temp/build.log'.
 * The ebuild environment file is located at '/tmp/portage/media-tv/mythtv-30.0_p20190808-r1/temp/environment'.
 * Working directory: '/tmp/portage/media-tv/mythtv-30.0_p20190808-r1/homedir'
 * S: '/tmp/portage/media-tv/mythtv-30.0_p20190808-r1/work/mythtv-30.0_p20190808-r1/mythtv'
```

thebitpit commented:
> Btw, to replace `egethome` and drop `user.eclass`, you can just check how the function is contructed from https://gitweb.gentoo.org/repo/gentoo.git/tree/eclass/user.eclass
> 
Is `egethome mythtv` wrong? I think it is more descriptive.
 
> But change it on next version bump. We'd like everyone to migrate to acct- method anyway.
> 
I thought I had already migrated with `DEPEND "acct-user/mythtv ...`

thebitpit commented:
> ```
> * ERROR: media-tv/mythtv-30.0_p20190808-r1::testworld failed (prepare phase):
>  *   The source directory '/tmp/portage/media-tv/mythtv-30.0_p20190808-r1/work/mythtv-30.0_p20190808-r1/mythtv' doesn't exist
>  ...

Fixed. 
I didn't notice that `S` used `PV` when I fixed the source code fetch to use `P`. It didn't matter until the ebuild was revision bumped when I reorganized commits.

juippis commented:
Thanks!
